scripts/generate_metadata.py

```diff
@@ -0,0 +1,236 @@
+    "fon": "Latn",
+    "ita": "Latn",
+    "nya": "Latn",
+    "aka": "Latn",
+    "por": "Latn",
+    "asm": "Beng",
+    "eus": "Latn",
+    "lin": "Latn",
+    "nep": "Deva",
+    "kin": "Latn",
+    "ind": "Latn",
+    "ara": "Arab",
+}
+
+
+def convert_code(code: str) -> str | None:
+    """Converts between two-letter and three-letter language codes"""
+    try:
+        lang_code = iso639.Language.match(code).part3
+        script = lang_to_script[lang_code]
+        return f"{lang_code}_{script}"
+    except Exception as e:
+        print(f"Couldn't convert {code}, reason: {e}")
+        return None
+
+
+api = HfApi()
+
+
+def get_embedding_dimensions(model_name: str) -> int | None:
+    try:
+        file_path = hf_hub_download(
+            repo_id=model_name, filename="1_Pooling/config.json"
+        )
+        with open(file_path) as in_file:
+            pooling_config = json.loads(in_file.read())
+            return pooling_config.get("word_embedding_dimension", None)
+    except Exception as e:
+        print(f"Couldn't get embedding size for {model_name}, reason: {e}")
+        return None
+
+
+def get_max_token(model_name: str) -> int | None:
+    try:
+        file_path = hf_hub_download(repo_id=model_name, filename="config.json")
+        with open(file_path) as in_file:
+            config = json.loads(in_file.read())
+            return config.get("max_position_embeddings", None)
+    except Exception as e:
+        print(f"Couldn't get embedding size for {model_name}, reason: {e}")
+        return None
+
+
+def model_meta_from_hf_hub(model_name: str) -> ModelMeta:
+    try:
+        card = ModelCard.load(model_name)
+        card_data = card.data.to_dict()
+        frameworks = ["PyTorch"]
+        if card_data.get("library_name", None) == "sentence-transformers":
+            frameworks.append("Sentence Transformers")
+        languages = card_data.get("language", None)
+        if isinstance(languages, str):
+            languages = [languages]
+        if languages is not None:
+            languages = [convert_code(l) for l in languages]
+            languages = [l for l in languages if l is not None]
+        repo_info = api.repo_info(model_name)
+        revision = repo_info.sha
+        release_date = repo_info.created_at.strftime("%Y-%m-%d")
+        try:
+            n_parameters = repo_info.safetensors.total
+        except Exception as e:
+            print(f"Couldn't get model size for {model_name}, reason: {e}")
+            n_parameters = None
+        n_dimensions = get_embedding_dimensions(model_name)
+        return ModelMeta(
+            name=model_name,
+            revision=revision,
+            release_date=release_date,
+            languages=languages,
+            license=card_data.get("license", None),
+            framework=frameworks,
+            n_parameters=n_parameters,
+            public_training_data=bool(card_data.get("datasets", None)),
+            adapted_from=None,
```

Comment:
Can be read from `config.json`. See `izhx/udever-bloom-560m` above for an example. The logic can be, if `_name_or_path` == `model_name` (with `/`), then adapted_from = None

Reply:
Thanks @isaac-chung this is very useful info<3